compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2345,6 +2348,136 @@ public void visit(BLangMatchGuard matchGuard) {
         matchGuard.expr.accept(this);
     }
 
+    @Override
+    public void visit(BLangMappingMatchPattern mappingMatchPattern) {
+        BRecordTypeSymbol recordSymbol =
+                Symbols.createRecordSymbol(0, names.fromString("$anonRecordType$" + recordCount++),
```

Comment:
Since this is a `VIRTUAL` symbol, it doesn't really matter. But I think as a best practice, it's better to treat any additional symbols that we create for desugared/inferred components/types just as we do for constructs in code and try to give as much as accurate details when creating the symbols (in this case, the `ANONYMOUS` flag).

@KavinduZoysa Also, any particular reason not to use `BLangAnonymousModelHelper` to generate the anon name?

Reply:
Fixed